### PR TITLE
Added samples for eDiscovery

### PR DESCRIPTION
**Pull request by mahage-msft**

### CODE REVIEW

File: sample-queries/sample-queries.json [hunk 1]
```diff
@@ -1679,6 +1679,65 @@
         "tip": "enable ExternalItem.Read.All",
         "skipTest": false
     },
+    {
+        "id": "26743d2a-2306-40f9-b3e8-3f857af69269",
+        "category": "Compliance (beta)",
+        "method": "GET",
+        "humanName": "List ediscoveryCases",
```
Comment (bettirosengugi) [inline sample-queries/sample-queries.json:1686]: Human name is the name users will see on GE to know what the query will do hence, should be in prose form e.g. List ediscovery cases

File: sample-queries/sample-queries.json [hunk 2]
```diff
@@ -1679,6 +1679,65 @@
...
+        "humanName": "Create reviewSetQuery",
+        "requestUrl": "/beta/compliance/ediscovery/cases/{caseId}/reviewSets/{reviewSetId}/queries",
+        "headers": [
+            {
+                "displayName":"Quarterly Financials Query",
+                "query":"(subject:\"Quarterly Financials\")"
+            }
+        ],
+        "docLink": "https://docs.microsoft.com/graph/api/reviewsetquery-post?view=graph-rest-beta",
+        "tip": "Replace {caseId} with the ID from an eDiscovery Case (List eDiscovery Cases) and {reviewSetId} with the ID from a review set that exists in that case (List reviewSets)",
```
Comment (bettirosengugi) [inline sample-queries/sample-queries.json:1738]: Give a query example for getting one of the id on the tip message so the user knows.

File: sample-queries/sample-queries.json [hunk 3]
```diff
@@ -1679,6 +1679,65 @@
...
+        "id": "26743d2a-2306-40f9-b3e8-3f857af69269",
+        "category": "Compliance (beta)",
+        "method": "GET",
+        "humanName": "List ediscoveryCases",
+        "requestUrl": "/beta/compliance/ediscovery/cases",
+        "docLink": "https://docs.microsoft.com/graph/api/ediscoverycase-list?view=graph-rest-beta",
+        "skipTest": false
+    },
+    {
+        "id": "d3f13cd9-2c89-4876-a88c-081e41370a02",
```
Comment (bettirosengugi) [inline sample-queries/sample-queries.json:1692]: Add a tip message to tell the user which permission they need to get the queries running.

File: sample-queries/sample-queries.json (hunk 1 above)
Comment (irvinesunday) [reply sample-queries/sample-queries.json:1686]: ... and usually starting with a lower case. See examples from the snippet below:

image

The underlined texts are what will be the `humanName`

File: sample-queries/sample-queries.json [hunk 4]
```diff
@@ -1679,6 +1679,65 @@
...
+        "category": "Compliance (beta)",
+        "method": "POST",
+        "humanName": "Create ediscoveryCase",
+        "requestUrl": "/beta/compliance/ediscovery/cases",
+        "headers": [
```
Comment (irvinesunday) [inline sample-queries/sample-queries.json:1697]: `headers` section will usually be of the format:

```
"headers": [
            {
                "name": "Content-type",
                "value": "application/json"
            }
        ]
```

File: sample-queries/sample-queries.json [hunk 5]
```diff
@@ -1679,6 +1679,65 @@
...
+            {
+                "displayName": "My first eDiscovery case"
```
Comment (irvinesunday) [inline sample-queries/sample-queries.json:1699]: This should ideally be the value of a `postBody` section, e.g.
```
"postBody":"{\r\n   \"displayName\": \"My first eDiscovery case\"  \r\n}"
```

**Comment by irvinesunday** [comment]
Arrangement of properties for a `POST` sample would follow the below sequence:
`id`
`category`
`method`
`humanName`
`requestUrl`
`docLink`
`headers`
`postBody`
`tip`
`skipTest`

File: sample-queries/sample-queries.json (hunk 2 above)
Comment (mahage-msft) [reply sample-queries/sample-queries.json:1738]: Can you point me to an example of what you mean?  There is a tip there and it references back to the APIs to use to get those on line 1738.  There's a bit of a conundrum here because getting the second ID requires the first ID. The tip would be a help topic by itself.

File: sample-queries/sample-queries.json (hunk 1 above)
Comment (mahage-msft) [reply sample-queries/sample-queries.json:1686]: I got the impression that the humanName was supposed to be the name that is visible in docs...

image

File: sample-queries/sample-queries.json (hunk 3 above)
Comment (mahage-msft) [reply sample-queries/sample-queries.json:1692]: I presume you're referring to the Graph Permissions like User.ReadAll.  Is that necessary since I don't think we even require that?  All permissioning is handled inside of compliance and Graph is simply a passthrough.  Should I link to something like this?  https://docs.microsoft.com/en-us/microsoft-365/compliance/assign-ediscovery-permissions?view=o365-worldwide.

File: sample-queries/sample-queries.json (hunk 4 above)
Comment (mahage-msft) [reply sample-queries/sample-queries.json:1697]: Gotcha, I mistook this for the postBody.  Fixed.

File: sample-queries/sample-queries.json (hunk 5 above)
Comment (mahage-msft) [reply sample-queries/sample-queries.json:1699]: Fixed.

File: sample-queries/sample-queries.json (hunk 2 above)
Comment (mahage-msft) [reply sample-queries/sample-queries.json:1738]: Fixed in all calls that have CaseID.  I appended the following text which I stole from another sample...

 To find the ID of an eDiscovery case, you can run: GET https://graph.microsoft.com/beta/compliance/ediscovery/cases.

There are some samples that also require another ID, I've left that out, hopefully they can figure it out once they see the patter.  If there's a more succinct way to express looking up and id and then using that to look up another ID, I'm all ears.

File: sample-queries/sample-queries.json (hunk 3 above)
Comment (mahage-msft) [reply sample-queries/sample-queries.json:1692]: This is a bit of an ongoing conversation with Irvine.  Would be nice if we can publish without any tips about permissions until we get to that.

**Comment by mahage-msft** [comment]
@irvinesunday - I rearranged the fields as you suggested above.

File: sample-queries/sample-queries.json (hunk 2 above)
Comment (bettirosengugi) [reply sample-queries/sample-queries.json:1741]: Sorry this wasnt clear, my suggestion is have the tip message as
Replace {caseId} with the ID from an eDiscovery Case (https://graph.microsoft.com/beta/compliance/ediscovery/cases) and {reviewSetId} with the ID from a review set that exists in that case (https://graph.microsoft.com/beta/compliance/ediscovery/cases/{caseid}/reviewSets)

File: sample-queries/sample-queries.json (hunk 3 above)
Comment (bettirosengugi) [reply sample-queries/sample-queries.json:1692]: For this, I meant add a tip value letting the user know the permissions required to run the query, but this is not important if we shall populate the Modify Permissions tab with the permissions required. but the tip would be:

`tip: "Go to Modify Permissions tab and consent to User.Read permission to run this query" `

Comment (mahage-msft) [reply sample-queries/sample-queries.json:1692]: I added the tip

File: sample-queries/sample-queries.json (hunk 2 above)
Comment (mahage-msft) [reply sample-queries/sample-queries.json:1741]: Gotcha, thanks for the clear example.  Incorporated in all examples where it makes sense.

**Review by bettirosengugi — APPROVED** [review]

**Comment by irvinesunday** [comment]
@mahage-msft I'm happy to approve as is, I just have a question...
Do you plan on adding GET, PATCH, DELETE samples for a single ediscoveryCase object?

**Comment by mahage-msft** [comment]
@irvinesunday - I was wondering about that.  Should I be adding samples for every API/method?  I tried to take the approach of just adding samples for the simple things.  Doing a PATCH requires a lot more explanation in the sample, more than just getting the appropriate IDs, you have to start adding a bunch of fields. Arguably, this isn't much different than creating a new case with a POST, which is included, see my next point.

For DELEETE, I'm a bit nervous about removing too much friction here.  DELETE can be horribly destructive and cannot be undone, this is by design because when cases need to be deleted, they need to be deleted with no trace.  There isn't a whole lot of friction in the UX to delete, but you typically have the people accessing the data who understand the consequences.  Removing too much friction here could be problematic, but I'm willing to entertain whatever you think.

Long story short, should I be adding samples for each and every API? Not a problem to do that, but would rather get these in and iterate before adding more.

**Comment by irvinesunday** [comment]
> @irvinesunday - I was wondering about that. Should I be adding samples for every API/method? I tried to take the approach of just adding samples for the simple things. Doing a PATCH requires a lot more explanation in the sample, more than just getting the appropriate IDs, you have to start adding a bunch of fields. Arguably, this isn't much different than creating a new case with a POST, which is included, see my next point.
> 
> For DELEETE, I'm a bit nervous about removing too much friction here. DELETE can be horribly destructive and cannot be undone, this is by design because when cases need to be deleted, they need to be deleted with no trace. There isn't a whole lot of friction in the UX to delete, but you typically have the people accessing the data who understand the consequences. Removing too much friction here could be problematic, but I'm willing to entertain whatever you think.
> 
> Long story short, should I be adding samples for each and every API? Not a problem to do that, but would rather get these in and iterate before adding more.

I hear you @mahage-msft. If a sample is aggressively destructive, I would advise against adding it in; a good example is this [permanent DELETE request for Applications](https://docs.microsoft.com/en-us/graph/api/directory-deleteditems-delete?view=graph-rest-1.0&tabs=http). This sample is not available in the list of samples in Graph Explorer. I think we can first merge and add these in and then later add in the PATCH sample. I however think we can also safely add the GET sample for a [single eDiscovery case object](https://docs.microsoft.com/en-us/graph/api/ediscoverycase-get?view=graph-rest-beta) right away (in the next PR perhaps). This would seem like a sample that would be highly used in this category; what do you think?

**Review by irvinesunday — APPROVED** [review]
👍🏽

**Comment by mahage-msft** [comment]
Fair enough, I'll add that in the next one since you already approved :).
